jdaviz/configs/cubeviz/plugins/viewers.py

```diff
@@ -37,6 +39,76 @@ def __init__(self, *args, **kwargs):
         self._initialize_toolbar_nested()
         self.state.add_callback('reference_data', self._initial_x_axis)
 
+        self.label_mouseover = None
+        self.add_event_callback(self.on_mouse_or_key_event, events=['mousemove', 'mouseenter',
+                                                                    'mouseleave'])
+
+    def on_mouse_or_key_event(self, data):
+
+        # Find visible layers
+        visible_layers = [layer for layer in self.state.layers if layer.visible]
+
+        if len(visible_layers) == 0:
+            return
+
+        if self.label_mouseover is None:
+            if 'g-coords-info' in self.session.application._tools:
+                self.label_mouseover = self.session.application._tools['g-coords-info']
+            else:
+                return
+
+        if data['event'] == 'mousemove':
+            # Display the current cursor coordinates (both pixel and world) as
+            # well as data values. For now we use the first dataset in the
+            # viewer for the data values.
+
+            # Extract first dataset from visible layers and use this for coordinates - the choice
+            # of dataset shouldn't matter if the datasets are linked correctly
+            image = visible_layers[0].layer
```

Comment:
how is `visible_layers` ordered?  If there are multiple layers and they're linked correctly, the coordinates shouldn't care what layer, but the _value_ will... is this guaranteed to be the "top" layer if multiple layers are plotted without changing transparency?  Is there any room to squeeze in the layer name for cases where `visible_layers` includes more than one?

This behavior/situation might be something that is worth adding test coverage, if possible.

Reply:
The inner workings of visible layers, I'd have to defer to @astrofrog or @maartenbreddels. I wish there is an easier way for me to reliably grab this info too but then Glue lets you overlay this and that, which complicates things. This particular info panel would ignore the semi-transparent overlay use cases.

Re: tests -- the info panel is kinda tested non-interactively in https://github.com/spacetelescope/jdaviz/blob/main/jdaviz/configs/imviz/tests/test_linking.py (the `label_mouseover` stuff). Not that the tests are exhaustive because they obviously missed #1299 but again adding a test for that would be out of scope here. Still, I guess I can add similar test(s) to Cubeviz...

Reply:
from trial and error, it seems that this does always show the "top" layer, which is the most sensible (simple) behavior.  If there is any room, I do think a label would be useful, but will approve this and that can always be added later if you don't have time to include it in scope here (as I think the same thing could apply to the existing display in imviz).

Reply:
Re: label -- In Imviz, I incorporated that into Compass plugin (because the label just seems to go with Compass naturally). Originally when I was new and naive, I attempted glue-viz/glue-jupyter#244 but failed. I don't think there is any more space for label in the coordinates display, as it is already pretty crowded.

This does bring up a good point... What if someone deselects the cube and have a 2D moment map displayed in the cube viewer instead? Is that a use case we even support?